### PR TITLE
Dusting Poly causes a power surge

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

With surges now added, causes a power surge event when Poly is dusted on the supermatter crystal.

Requires https://github.com/tgstation/tgstation/pull/78492

## Why It's Good For The Game

Adds some flavor to killing the engineering department's pet.

## Changelog

:cl: LT3
add: Poly now causes a power surge when dusted by the supermatter crystal
/:cl:

### CODE REVIEW

File: code/modules/events/supermatter_surge.dm [hunk 1]
```diff
@@ -99,6 +99,24 @@
 	engine = null
 	sm_gas = null
 
+/datum/round_event_control/supermatter_surge/poly
+	name = "Supermatter Surge: Poly's Revenge"
+	typepath = /datum/round_event/supermatter_surge/poly
+	category = EVENT_CATEGORY_ENGINEERING
+	weight = 0
```
Comment (Rhials) [inline code/modules/events/supermatter_surge.dm:106]: Make sure the max_occurrences is 0 too. Events with 0 weight can still trigger naturally (or so I've heard). 

Events that are being forced don't check max_occurrences, so it wouldn't hurt to add the extra layer of assurance.

Comment (lessthnthree) [reply code/modules/events/supermatter_surge.dm:106]: Ahh, sure.

File: code/modules/events/supermatter_surge.dm [hunk 2]
```diff
@@ -80,8 +80,8 @@
 
 	end_when = rand(SURGE_DURATION_MIN, SURGE_DURATION_MAX)
 
-/datum/round_event/supermatter_surge/announce()
-	priority_announce("The Crystal Integrity Monitoring System has detected unusual atmospheric properties in the supermatter chamber and energy output from the supermatter crystal has increased significantly. Engineering intervention is required to stabilize the engine.", "Class [surge_class] Supermatter Surge Alert", 'sound/machines/engine_alert3.ogg')
+/datum/round_event/supermatter_surge/announce(fake)
+	priority_announce("The Crystal Integrity Monitoring System has detected unusual atmospheric properties in the supermatter chamber, energy output from the supermatter crystal has increased significantly. Engineering intervention is required to stabilize the engine.", "Class [surge_class] Supermatter Surge Alert", 'sound/machines/engine_alert3.ogg')
```
Comment (Dorsisdwarf) [inline code/modules/events/supermatter_surge.dm:84]: Why replace the and with a comma, isn't that less correct?

**Comment by deathrobotpunch** [comment]
poly’s revenge

File: code/modules/events/supermatter_surge.dm (hunk 2 above)
Comment (Jacquerel) [reply code/modules/events/supermatter_surge.dm:84]: they're both just about as correct as each other

Comment (lessthnthree) [reply code/modules/events/supermatter_surge.dm:84]: It was originally CIMS as an acronym, but now that the full name is used the sentence was feeling a little long without a pause.

**Review by Jacquerel — APPROVED** [review]

**Comment by ZephyrTFA** [comment]
The parrot is dead, long live the parrot

File: code/modules/events/supermatter_surge.dm (hunk 2 above)
Comment (Dorsisdwarf) [reply code/modules/events/supermatter_surge.dm:84]: a comma is not a direct substitute for "and" but I suppose its a vanishingly minor quibble

Comment (Jacquerel) [reply code/modules/events/supermatter_surge.dm:84]: this sentence is completely grammatically correct with either "and" or a comma.

**Pull request merged**